src/macros/firstOrFail.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+use Illuminate\Support\Collection;
+
+/*
+ * Get the first item. Throws Exception when collection is empty.
+ *
+ * @return mixed
+ */
+Collection::macro('firstOrFail', function () {
+    if (! is_null($item = $this->first())) {
+        return $item;
+    }
+
+    throw new \Exception('No items found in collection.');
```

Comment:
Let's throw a dedicated exception Instead of throwing a generic one. This way people can add custom behaviour in the exception handler. Name that exception:
`Spatie\CollectionMacros\Exceptions\CollectionItemNotFound`